[PATCH] usb-storage: Pentax K10D reports wrong capacity

Jaak Ristioja reported problems with his Pentax K10D camera:
https://bugs.gentoo.org/show_bug.cgi?id=250406

The number of reported sectors is off-by-one.

Signed-off-by: Daniel Drake <dsd@gentoo.org>

Index: linux-2.6.27-gentoo-r4/drivers/usb/storage/unusual_devs.h
===================================================================
--- linux-2.6.27-gentoo-r4.orig/drivers/usb/storage/unusual_devs.h
+++ linux-2.6.27-gentoo-r4/drivers/usb/storage/unusual_devs.h
@@ -1363,6 +1363,13 @@ UNUSUAL_DEV( 0x0a17, 0x006, 0x0000, 0xff
 		US_SC_DEVICE, US_PR_DEVICE, NULL,
 		US_FL_FIX_INQUIRY ),
 
+/* Reported by Jaak Ristioja <Ristioja@gmail.com> */
+UNUSUAL_DEV( 0x0a17, 0x006e, 0x0100, 0x0100,
+		"Pentax",
+		"K10D",
+		US_SC_DEVICE, US_PR_DEVICE, NULL,
+		US_FL_FIX_CAPACITY ),
+
 /* These are virtual windows driver CDs, which the zd1211rw driver
  * automatically converts into WLAN devices. */
 UNUSUAL_DEV( 0x0ace, 0x2011, 0x0101, 0x0101,
